### DISCUSSION TITLE
Added: Notes for October 7th, 2020

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#3983] [LDM Notes]
https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-10-07.md

## Agenda

- `record struct` syntax
- `data` members redux
- `ReadOnlySpan<char>` patterns

**Comment by orthoxerox** [comment]
Really happy about `ReadOnlySpan<char>`.

Coming back to readonly structs, what's the proposed syntax for making a readonly record struct? `public readonly record struct Point3D(float X, float Y, float Z);`?

**Reply by 333fred** [reply]
Yep. We also talked about variants where you could put readonly on the individual parameters if you don't want the whole thing to be immutable.

**Reply by HaloFour** [reply]
Sounds like that dovetails with readonly parameters in general.

**Comment by spydacarnage** [comment]
I would assume that as, in the agreed version, `record` would be a modifier of `struct`, then `record` and `readonly` could co-exist, so it makes sense that `readonly record struct` would be the way.
